Makefile.am

```diff
@@ -2,7 +2,27 @@ SUFFIXES= .rst
 
 EXTRA_DIST = COPYING scripts/findstatic.pl swupd.bash
 
-AM_CFLAGS = -fPIC -O2 -g -Wall -W -Wformat-security -D_FORTIFY_SOURCE=2 -fno-common -std=gnu99
+AM_CFLAGS = -fPIC -O2 -g -D_FORTIFY_SOURCE=2 -fno-common \
+	-fstack-protector-strong -fstack-clash-protection -fcf-protection -fpie -std=gnu99 \
+	-Wall \
+	-Wextra \
+	-Wfatal-errors \
+	-Werror \
```

Comment:
I wouldn't add -Werror here. We add -Werror on travis to break the tests if any warning is present, but as a developer I prefer to see errors than warnings.

-Wfatar-errors is not nice for a dev too. I would take this out